tests/test_config.py

```diff
@@ -348,54 +347,39 @@ class Model(BaseModel):
         assert m == m.model_copy()
 
     def test_config_class_is_deprecated(self):
-        with warnings.catch_warnings():
-            # we need to explicitly ignore the other warning in pytest-8
-            # TODO: rewrite it to use two nested pytest.warns() when pytest-7 is no longer supported
-            warnings.simplefilter('ignore')
-            with pytest.warns(
-                PydanticDeprecatedSince20,
-                match='Support for class-based `config` is deprecated, use ConfigDict instead.',
-            ):
-
-                class Config(BaseConfig):
-                    pass
+        with pytest.warns(PydanticDeprecatedSince20) as all_warnings:
+
+            class Config(BaseConfig):
+                pass
+
+        assert len(all_warnings) == 2
+        expected_warnings = [
+            'BaseConfig is deprecated. Use the `pydantic.ConfigDict` instead',
+            'Support for class-based `config` is deprecated, use ConfigDict instead',
+        ]
+        assert [w.message.message for w in all_warnings] == expected_warnings
 
     def test_config_class_attributes_are_deprecated(self):
```

Comment:
Looks like we're getting some failures for these two:

https://github.com/pydantic/pydantic/actions/runs/9590698692/job/26446460308?pr=9702

The changes look alright to me, though I haven't dug into the error cause.

Reply:
Would this be a case where the nested context managers could help?

Reply:
In some ways, I'd rather we go with that approach so that the error messages are localized and make more sense when contextualized directly above the error-throwing statement.

Reply:
I doubt this one can be nested -- since `typing-extensions=4.6.1` only emits one warning rather two, there'd still be a failure when the outer block doesn't throw a warning.